bliss/surveys/sdss.py

```diff
@@ -507,63 +508,44 @@ def __init__(
         self, survey_data_dir, image_ids, image_items, bands, ref_band, prior_config: PriorConfig
     ):
         super().__init__(bands, **prior_config)
-        self.stars_fluxes, self.gals_fluxes = self._flux_ratios_against_b(
-            survey_data_dir, image_ids, image_items, b=ref_band
-        )
-
-    def _flux_ratios_against_b(self, survey_data_dir, image_ids, items, b) -> Tuple[dict, dict]:
-        stars_fluxes = {}
-        gals_fluxes = {}
+        self.stars_fluxes, self.gals_fluxes = self._flux_ratios_against_b(survey_data_dir, ref_band)
```

Comment:
Rather than drawing samples in the constructor, could the prior distribution have a function called `sample(...)`, taking `num_samples` as an argument?

Reply:
Like, it makes sense for the constructor to load the `pkl` files. But you wouldn't want to redo it every time you generate a batch. You'd want to do it once, keep the `SDSSPrior` object in memory, and reuse it every time you generate a batch. Is that what's happening currently?

Reply:
^ I agree with that overall approach. But I think the current code doesn't even redo it each batch - `ImagePrior` only gets instantiated once, and `_flux_ratios_against_b` only samples `batch_size` ratios. So I think each batch is going to use the same set of ratios instead of resampling each time.

Reply:
It does not redo the ratios every batch - fixed!